src/api/utils/query-parser/index.js

```diff
@@ -0,0 +1,225 @@
+        filter.$and[i].$or[j][keyMap[q[i][j].key].value] = query
+        if (keyMap[q[i][j].key].relativeValue) {
+          response.mapReduce = true
+        }
+      }
+    }
+
+    response.tokens = q
+    response.filter = filter
+    return response
+  }
+
+  /*
+    Split query into individual tokens.
+  */
+  static tokenize (query) {
+    let pieces = query.split(' AND ')
+
+    for (let i = 0; i < pieces.length; i++) {
+      pieces[i] = pieces[i].trim().split(' OR ')
+      for (let j = 0; j < pieces[i].length; j++) {
+        pieces[i][j] = pieces[i][j].trim()
+      }
+    }
+
+    return pieces
+  }
+
+  /*
+    Parse token to retrieve key and filter information.
+  */
+  static parseToken (token) {
+    let response = {}
+
+    response = {
+      raw: token,
+      key: token.slice(0, token.indexOf(':')).trim(),
+      filter: token.slice(token.indexOf(':') + 1).trim()
+    }
+
+    if (response.filter.indexOf('-') === 0) {
+      // Negation
+      response.filter = {
+        operator: '-',
+        value: response.filter.substring(1)
+      }
+    } else if (response.filter.indexOf('>=') === 0) {
+      response.filter = {
+        operator: '>=',
+        value: response.filter.substring(2)
+      }
+    } else if (response.filter.indexOf('<=') === 0) {
+      response.filter = {
+        operator: '<=',
+        value: response.filter.substring(2)
+      }
+    } else if (response.filter.indexOf('>') === 0) {
+      response.filter = {
+        operator: '>',
+        value: response.filter.substring(1)
+      }
+    } else if (response.filter.indexOf('<') === 0) {
+      response.filter = {
+        operator: '<',
+        value: response.filter.substring(1)
+      }
+    } else {
+      response.filter = {
+        operator: undefined,
+        value: response.filter
+      }
+    }
+
+    if (isNaN(parseFloat(response.filter.value)) || !isFinite(response.filter.value)) {
+      // Is not a number
+      response.filter.value = response.filter.value.substring(1,
+        response.filter.value.length - 1)
+    } else {
+      response.filter.value = parseFloat(response.filter.value)
+    }
+
+    return response
+  }
+
+  /*
+    Form the MongoDB-compatible query for strings.
+  */
+  static stringQuery (filter) {
+    if (filter.operator === '-') {
+      return {
+        $regex: '^((?!' + escapeRe(filter.value) + ').)*$',
+        $options: 'i'
+      }
+    }
+    return { $regex: '(?i).*' + escapeRe(filter.value) + '.*' }
+  }
+
+  /*
+    Form the MongoDB-compatible query for numbers.
+  */
+  static numberQuery (filter) {
+    if (filter.operator === '>') {
+      return { $gt: filter.value }
+    } else if (filter.operator === '<') {
+      return { $lt: filter.value }
+    } else if (filter.operator === '>=') {
+      return { $gte: filter.value }
+    } else if (filter.operator === '<=') {
+      return { $lte: filter.value }
+    } else if (filter.operator === '-') {
+      let newValue = -filter.value
```

Comment:
Just realized that the food API doesn't use the new parser yet. I tried it with buildings, same thing happens:

+ `/1.0/buildings/filter?q=lat:43.66319655917286` looks for `lat = 43.66319655917286`
+ `/1.0/buildings/filter?q=lat:-43.66319655917286` looks for `lat = -43.66319655917286`

and

+ `/1.0/buildings/filter?q=lng:-79.395819910554` looks for `lng = -79.395819910554`
+ `/1.0/buildings/filter?q=lng:--79.395819910554` looks for `lng = 79.395819910554`